File: 1463.The K Weakest Rows in a Matrix/solution.py
class Solution(object):
    def kWeakestRows(self, mat, k):
        """
        :type mat: List[List[int]]
        :type k: int
        :rtype: List[int]
        """
        rec = {}
        n = len(mat)
        for i in range(n): rec[i] = sum(mat[i])
        # print(rec)
        temp = sorted(rec.items(),key = lambda x:x[1])
        # print(temp)
        return [r[0] for r in temp[:k]]

File: 1463.The K Weakest Rows in a Matrix/solution.cpp
class Solution {
public:
    vector<int> kWeakestRows(vector<vector<int>>& mat, int k) {
        int n = mat.size();
        int m = mat[0].size();

        vector<int> row(n);
        // int ind = 0;
        for(int i=0; i<n; i++){
            int c = 0;
            for(int j=0; j<m; j++){
                if(mat[i][j]==1){
                    c++;
                }
            }
            row[i] = c;
        }
        vector<int> ind(n);
        for(int i=0; i<n; i++){
            ind[i] = i;
        }
        sort(ind.begin(), ind.end(), [&](int a, int b){
            if(row[a] != row[b]) {
                return row[a] < row[b];
            }
            return a < b;
            // return row[a]<row[b];
        });
        vector<int> res(k);
        for(int i=0; i<k; i++){
            res[i] = ind[i];
        }
        return res;
    }
};